android/src/main/kotlin/co/doneservices/callkeep/IncomingCallActivity.kt

```diff
@@ -0,0 +1,299 @@
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_BACKGROUND_URL
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_ACCENT_COLOR
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_DURATION
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_INCOMING_DATA
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_CALLER_NAME
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_CONTENT_TITLE
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_HANDLE
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_HEADERS
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_LOGO
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_HAS_VIDEO
+import com.squareup.picasso.Picasso
+import de.hdodenhof.circleimageview.CircleImageView
+import kotlin.math.abs
+import okhttp3.OkHttpClient
+import com.squareup.picasso.OkHttp3Downloader
+import android.view.ViewGroup.MarginLayoutParams
+import android.os.PowerManager
+import android.os.PowerManager.WakeLock
+import android.text.TextUtils
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_ACCEPT_TEXT
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_DECLINE_TEXT
+
+
+class IncomingCallActivity : Activity() {
+
+    companion object {
+
+        const val ACTION_ENDED_CALL_INCOMING =
+                "co.doneservices.callkeep.ACTION_ENDED_CALL_INCOMING"
+
+        fun getIntent(context: Context, data: Bundle) = Intent(ACTION_CALL_INCOMING).apply {
+            action = "${context.packageName}.${ACTION_CALL_INCOMING}"
+            putExtra(EXTRA_CALLKEEP_INCOMING_DATA, data)
+            flags =
+                    Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_REORDER_TO_FRONT
+        }
+
+        fun getIntentEnded(context: Context) =
+                Intent("${context.packageName}.${ACTION_ENDED_CALL_INCOMING}")
+
+    }
+
+    inner class EndedCallKeepBroadcastReceiver : BroadcastReceiver() {
+        override fun onReceive(context: Context?, intent: Intent?) {
+            if (!isFinishing) {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                    finishAndRemoveTask()
+                } else {
+                    finish()
+                }
+            }
+        }
+    }
+
+    private var endedCallKeepBroadcastReceiver = EndedCallKeepBroadcastReceiver()
+
+    private lateinit var llBackground: LinearLayout
+
+    private lateinit var tvCallerName: TextView
+    private lateinit var tvCallHeader: TextView
+    private lateinit var ivLogo: ImageView
+
+    private lateinit var btnAnswer: Button
+
+    private lateinit var btnDecline: Button
```

Comment:
Is Hungarian notation common in Android?

Reply:
Yes